t/24-worker-overall.t

```diff
@@ -211,81 +212,126 @@ subtest 'accept or skip next job' => sub {
     subtest 'get next job in queue' => sub {
         my @pending_jobs = (0, [1, [2, 3], [4, 5]], 6, 7, [8, 9, [10, 11]], 12);
         my @expected_iteration_steps = (
-            [0, [[1, [2, 3], [4, 5]], 6, 7, [8, 9, [10, 11]], 12]],
-            [1, [[2, 3], [4, 5]]],
-            [2, [3]],
-            [3, []],
-            [4, [5]],
-            [5, []],
-            [6, [7, [8, 9, [10, 11]], 12]],
-            [7, [[8, 9, [10, 11]], 12]],
-            [8, [9, [10, 11]]],
-            [9, [[10, 11]]],
-            [10, [11]],
-            [11, []],
-            [12, []],
-            [undef, []],
+            # expected [next job id, current chain, parent chain]
+            [0, [[1, [2, 3], [4, 5]], 6, 7, [8, 9, [10, 11]], 12], [0]],
+            [1, [[2, 3], [4, 5]], [0, 1]],
+            [2, [3], [0, 1, 2]],
+            [3, [], [0, 1, 2]],
+            [4, [5], [0, 1, 4]],
+            [5, [], [0, 1, 4]],
+            [6, [7, [8, 9, [10, 11]], 12], [0]],
+            [7, [[8, 9, [10, 11]], 12], [0]],
+            [8, [9, [10, 11]], [0, 8]],
+            [9, [[10, 11]], [0, 8]],
+            [10, [11], [0, 8, 10]],
+            [11, [], [0, 8, 10]],
+            [12, [], [0]],
```

Comment:
Each row now contains the relevant parents for the current iteration step. Otherwise everything is the same.